lib/note-list/index.tsx

```diff
@@ -59,244 +52,126 @@ type DispatchProps = {
+type NoteListItem =
+  | T.NoteEntity
+  | 'tag-suggestions'
+  | 'notes-header'
+  | 'no-notes';
+
+const heightCache = new CellMeasurerCache({
+  // row height base is 21px for the title + 18px vertical padding
+  // max preview lines is 4 lines of 24px
+  defaultHeight: 21 + 18 + 24 * 4,
+  fixedWidth: true,
+});
 
 /**
  * Renders an individual row in the note list
  *
  * @see react-virtual/list
  *
- * @param {Object[]} notes list of filtered notes
- * @param {String} searchQuery search searchQuery
- * @param {String} noteDisplay list view style: comfy, condensed, expanded
- * @param {Number} selectedNoteId id of currently selected note
- * @param {Function} onSelectNote used to change the current note selection
- * @param {Function} onPinNote used to pin a note to the top of the list
- * @returns {Function} does the actual rendering for the List
+ * @param notes list of filtered notes
+ * @param searchQuery search searchQuery
+ * @param noteDisplay list view style: comfy, condensed, expanded
+ * @param selectedNoteId id of currently selected note
+ * @param onSelectNote used to change the current note selection
+ * @param onPinNote used to pin a note to the top of the list
+ * @param isSmallScreen whether we're in a narrow view
+ * @returns does the actual rendering for the List
  */
 const renderNote = (
-  notes,
+  notes: NoteListItem[],
   {
     searchQuery,
     noteDisplay,
     selectedNoteId,
     onSelectNote,
     onPinNote,
     isSmallScreen,
+  }: {
+    searchQuery: string;
+    noteDisplay: T.ListDisplayMode;
+    selectedNoteId?: T.EntityId;
+    onSelectNote: DispatchProps['onSelectNote'];
+    onPinNote: DispatchProps['onPinNote'];
+    isSmallScreen: boolean;
   }
-) => ({ index, rowIndex, key, style }) => {
-  const note = notes['undefined' === typeof index ? rowIndex : index];
-
-  // handle special sections
-  switch (note) {
-    case 'notes-header':
-      return (
-        <div key={key} style={style} className="note-list-header">
-          Notes
-        </div>
-      );
-    case 'tag-suggestions':
-      return (
-        <div key={key} style={style}>
+): ListRowRenderer => ({ index, key, parent, style }) => {
+  const note = notes[index];
+
+  if (
+    'tag-suggestions' === note ||
+    'notes-header' === note ||
+    'no-notes' === note
+  ) {
+    return (
+      <CellMeasurer
+        cache={heightCache}
+        columnIndex={0}
+        key={key}
+        parent={parent}
+        rowIndex={index}
+      >
+        {'tag-suggestions' === note ? (
           <TagSuggestions />
-        </div>
-      );
-    case 'no-notes':
-      return (
-        <div key={key} style={style} className="note-list is-empty">
-          <span className="note-list-placeholder">No Notes</span>
-        </div>
-      );
+        ) : 'notes-header' === note ? (
+          <div className="note-list-header">Notes</div>
+        ) : (
+          <div className="note-list is-empty">
```

Comment:
I think we need to keep the style argument here so that the empty note list gets the correct height? (But just trying that by itself did not fix the issue)

Reply:
So I know what the fix for this *should* be, and it's:

```
  style={{  ...style, height: 200 }}
```
But that yields THIS nonsense, for which I have no explanation whatsoever except that I clearly do not understand anything:

<img width="328" alt="Screen Shot 2020-04-13 at 10 34 30 PM" src="https://user-images.githubusercontent.com/52152/79189658-2e830300-7dd7-11ea-9942-603b970b515f.png">